packages/web-components/src/theme/set-theme.ts

```diff
@@ -10,18 +10,23 @@ const tokenNames = Object.keys(tokens) as (keyof Theme)[];
  */
 export const setTheme = (theme: Theme) => {
   for (const t of tokenNames) {
+    let registered = false;
+
     if ('registerProperty' in CSS) {
       try {
         CSS.registerProperty({
           name: `--${t}`,
           inherits: true,
           initialValue: theme[t] as string,
         });
-      } catch {
-        document.body.style.setProperty(`--${t}`, theme[t] as string);
-      }
-    } else {
-      document.body.style.setProperty(`--${t}`, theme[t] as string);
+        registered = true;
+      } catch {}
+    }
+
+    if (!registered) {
+      // TODO: Maybe find a better way to update the values. Current approach
+      // adds lots of code to the `style` attribute on `<body>`.
```

Comment:
Good call here - we should also elevate this to the Edge web plat team as this seems like a gap for super common behavior. One thing we've looked at is also using an adoptedStyleSheet to set this as it reduced calls to the document down to one. That can be a further out investigation :)

Reply:
Yup! I started looking into that but wanted to keep this change small. Will looking into that afterward!